### PR TITLE
chore: apply latest reducer proto to the reducer SDK 

### PR DESCRIPTION
**Pull request by KeranYang**

### CODE REVIEW

File: src/main/java/io/numaproj/numaflow/reducer/ReduceActor.java [hunk 1]
```diff
@@ -44,23 +44,34 @@ private void invokeHandler(HandlerDatum handlerDatum) {
...
+        responseBuilder.setWindow(ReduceOuterClass.Window.newBuilder()
+                .setStart(Timestamp.newBuilder()
+                        .setSeconds(this.md.getIntervalWindow().getStartTime().getEpochSecond())
+                        .setNanos(this.md.getIntervalWindow().getStartTime().getNano()))
+                .setEnd(Timestamp.newBuilder()
+                        .setSeconds(this.md.getIntervalWindow().getEndTime().getEpochSecond())
+                        .setNanos(this.md.getIntervalWindow().getEndTime().getNano()))
+                .setSlot("slot-0").build());
+        // if we start building the response, it means we already reached EOF.
+        responseBuilder.setEOF(true);
```
Comment (KeranYang) [inline src/main/java/io/numaproj/numaflow/reducer/ReduceActor.java:64]: fix this

File: src/main/proto/reduce/v1/reduce.proto [hunk 2]
```diff
@@ -19,22 +19,58 @@ service Reduce {
...
+  google.protobuf.Timestamp start = 1;
+  google.protobuf.Timestamp end = 2;
+  string slot = 3;
 }
 
 /**
  * ReduceResponse represents a response element.
  */
 message ReduceResponse {
+  // FIXME: put all fields(window, EOF) inside of Result. Reference: https://protobuf.dev/programming-guides/api/#dont-include-primitive-types
```
Comment (KeranYang) [inline src/main/proto/reduce/v1/reduce.proto:59]: create an issue for it

Comment (KeranYang) [reply src/main/proto/reduce/v1/reduce.proto:59]: https://github.com/numaproj/numaflow-go/issues/96

File: src/main/java/io/numaproj/numaflow/reducer/ReduceSupervisorActor.java [hunk 3]
```diff
@@ -119,24 +117,26 @@ private void responseListener(ActorResponse actorResponse) {
...
-            responseObserver.onCompleted();
-            getContext().getSystem().stop(getSelf());
+        String uniqueId = actorResponse.getUniqueIdentifier();
+        if (actorResponse.getResponse().getEOF()) {
+            actorsMap.remove(uniqueId);
+            if (actorsMap.isEmpty()) {
+                responseObserver.onCompleted();
+                getContext().getSystem().stop(getSelf());
+            }
         }
```
Comment (yhl25) [inline src/main/java/io/numaproj/numaflow/reducer/ReduceSupervisorActor.java:128]:
```suggestion
        if (actorResponse.getResponse().getEOF()) {
            actorsMap.remove(actorResponse.getUniqueIdentifier(););
            if (actorsMap.isEmpty()) {
                responseObserver.onCompleted();
                getContext().getSystem().stop(getSelf());
            }
        }
```

**Review by yhl25 — APPROVED** [review]

**Pull request merged**